stl/inc/stop_token

```diff
@@ -0,0 +1,401 @@
+// stop_token standard header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _STOP_TOKEN_
+#define _STOP_TOKEN_
+#include <yvals.h>
+#if _STL_COMPILER_PREPROCESSOR
+
+#if !_HAS_CXX20
+#pragma message("The contents of <stop_token> are available only with C++20 or later.")
+#else // ^^^ !_HAS_CXX20 / _HAS_CXX20 vvv
+
+#include <atomic>
+#include <xmemory>
+#include <xthreads.h>
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+_STD_BEGIN
+struct nostopstate_t {
+    explicit nostopstate_t() = default;
+};
+
+inline constexpr nostopstate_t nostopstate{};
+
+struct _Stop_state;
+class stop_token;
+
+class _Stop_callback_base {
+    friend _Stop_state;
+
+public:
+    using _Callback_fn = void(__cdecl*)(_Stop_callback_base*) _NOEXCEPT_FNPTR;
+
+    explicit _Stop_callback_base(const _Callback_fn _Fn_) noexcept : _Fn{_Fn_} {}
+    _Stop_callback_base(const _Stop_callback_base&) = delete;
+    _Stop_callback_base(_Stop_callback_base&&)      = delete;
+    _Stop_callback_base& operator=(const _Stop_callback_base&) = delete;
+    _Stop_callback_base& operator=(_Stop_callback_base&&) = delete;
+
+    // if _Token is _Stop_requested, calls the callback;
+    // otherwise, inserts *this into the callback list if stop is possible
+    inline void _Attach(const stop_token& _Token) noexcept;
+    inline void _Attach(stop_token&& _Token) noexcept;
+
+    // if *this is in a callback list, removes it
+    inline void _Detach() noexcept;
+
+private:
+    template <bool _Transfer_ownership>

```

Comment:
I don't think so; it would make `_Do_attach` less clear (since it now says `if constexpr (_Transfer_ownership)`) and the only two callers are the two _Attach overloads which are right next to each other near 302.